### PR TITLE
Rte state

### PR DESCRIPTION
**Pull request by adriangroch**
Adds the proposed idea from #2336

### CODE REVIEW

File: src/rich_text_editor/model/RichTextEditor.js [hunk 1]
```diff
@@ -156,11 +195,13 @@ export default class RichTextEditor {
    */
   syncActions() {
     this.getActions().forEach(action => {
-      const event = action.event || 'click';
-      action.btn[`on${event}`] = e => {
-        action.result(this, action);
-        this.updateActiveActions();
-      };
+      if (!action.state || (action.state && action.state() >= 0)) {
```
Comment (artf) [inline src/rich_text_editor/model/RichTextEditor.js]: I think you should call the `state` function with the proper signature `state(rte, doc)`

**Comment by adriangroch** [comment]
> Cool 👍, just check the comment I've left you

Done :)

**Comment by artf** [comment]
@adriangroch seems like the change breaks the tests 🤔

**Comment by adriangroch** [comment]
I _think_ it needed to ensure the `actionBar` existed first, added a check to ensure that.

**Comment by artf** [comment]
Great, thanks @adriangroch for the contribution

**Pull request merged**